src/http/metrics.rs

```diff
@@ -0,0 +1,802 @@
+            "number of currently open RTR connections",
+            MetricType::Gauge
+        ),
+        metrics.current_connections()
+    );
+    target.single(
+        Metric::new(
+            "rtr_bytes_read",
+            "total number of bytes read from RTR connections",
+            MetricType::Counter
+        ),
+        metrics.bytes_read()
+    );
+    target.single(
+        Metric::new(
+            "rtr_bytes_written",
+            "total number of bytes weitten to RTR connections",
+            MetricType::Counter
+        ),
+        metrics.bytes_written()
+    );
+
+    if detailed {
+        let item = Metric::new(
+            "rtr_client_connections",
+            "number of currently open connections per client address",
+            MetricType::Gauge
+        );
+        target.header(item);
+        metrics.fold_clients(0, |count, client| {
+            if client.is_open() {
+                *count += 1
+            }
+        }).for_each(|(addr, count)| {
+            target.multi(item).label("addr", addr).value(count)
+        });
+
+        let item = Metric::new(
+            "rtr_client_serial",
+            "last serial seen by a client address",
+            MetricType::Gauge
+        );
+        target.header(item);
+        metrics.fold_clients(None, |serial, client| {
+            *serial = match (*serial, client.serial().map(u32::from)) {
+                (Some(left), Some(right)) => Some(cmp::max(left, right)),
+                (Some(left), None) => Some(left),
+                (None, Some(right)) => Some(right),
+                (None, None) => None
+            };
+        }).for_each(|(addr, count)| {
+            match count {
+                Some(count) => {
+                    target.multi(item).label("addr", addr).value(count);
+                }
+                None => {
+                    target.multi(item).label("addr", addr).value(-1);
+                }
+            }
+        });
+
+        let item = Metric::new(
+            "rtr_client_last_update_seconds",
+            "seconds since last update by a client address",
+            MetricType::Gauge
+        );
+        target.header(item);
+        metrics.fold_clients(None, |update, client| {
+            *update = match (*update, client.updated()) {
+                (Some(left), Some(right)) => Some(cmp::max(left, right)),
+                (Some(left), None) => Some(left),
+                (None, Some(right)) => Some(right),
+                (None, None) => None
+            };
+        }).for_each(|(addr, update)| {
+            match update {
+                Some(update) => {
+                    let duration = Utc::now() - update;
+                    target.multi(item).label("addr", addr).value(
+                        format_args!(
+                            "{}.{:03}",
+                            duration.num_seconds(),
+                            duration.num_milliseconds() % 1000,
```

Comment:
`subsec_millis` for consistency?

Reply:
This is actually a `chrono::Duration` that doesn’t have that method.